spec.emu

```diff
@@ -500,45 +580,145 @@ location: https://github.com/tc39/proposal-decimal/
     <emu-alg>
-      1. Let _roundingMode_ be ? Get(_options_, *"roundingMode"*).
-      1. If _roundingMode_ is *undefined*, set _roundingMode_ to the <emu-xref href="#dfn-decimal128-default-rounding-mode">default rounding mode</emu-xref>.
-      1. If _roundingMode_ is not a <emu-xref href="#dfn-decimal128-rounding-mode">rounding mode</emu-xref>, throw a *RangeError* exception.
-      1. Let _format_ be ? Get(_options_, *"format*").
-      1. If _format_ is *undefined*, set _format_ to *"decimal"*.
-      1. If _format_ is not in « *"decimal"*, *"exponential"* », then throw a *RangeError* exception.
-      1. If _format_ is *"decimal*", then
-        1. Return Decimal128ToDecimalString(*this*).
-      1. Return Decimal128ToExponentialString(*this*).
+      1. If _precision_ is neither *undefined* nor a Number, throw a *TypeError* exception.
+      1. Let _O_ be the *this* value.
+      1. Perform ? RequireInternalSlot(_O_, [[Decimal128Data]]).
+      1. If _precision_ is *undefined_, return Decimal128ToDecimalString(_O_).
+      1. If _precision_ is not an integer, throw a *RangeError* exception.
+      1. If _precision_ < 1, throw a *RangeError* exception.
+      1. Let _d_ be _O_.[[Decimal128Data]].
+      1. If _d_ is *NaN*<sub>𝔻</sub>, return *"NaN"*.
+      1. If _d_ is *+∞*<sub>𝔻</sub>, return *"Infinity"*.
+      1. If _d_ is *-∞*<sub>𝔻</sub>, return *"-Infinity"*.
+      1. Let _v_ be _d_[0].
+      1. Let _q_ be _d_[1].
+      1. If _v_ is *+0*<sub>𝔻</sub>, return *"0"*.
+      1. If _v_ is *-0*<sub>𝔻</sub>, return *"-0"*.
+      1. If -_q_ < _precision_, then
+        1. Let _s_ be Decimal128ToDecimalString(_O_).
+        1. Let _integerDigits_ be the substring of _s_ up to the first occurrence of *"."* in _s_.
+        1. Let _nonIntegerDigits_ be the substring of _s_ after the first occurrence of *"."* in _s_.
+        1. Let _numTrailingZeroesToAdd_ be _precision_ minus the length of _nonIntegerDigits_.
+        1. Let _trailingZeroes_ be the string *"0"* repeated _numTrailingZeroesToAdd_ times.
+        1. Return the concatenation of _integerDigits_, *"."*, _nonIntegerDigits_, and _trailingZeroes_.
+      1. Let _n_ be the integer such that v × 10<sup>-_q_</sup> = _n_ and 0 < abs(_n_) < 10<sup>34</sup>.
+      1. Let _n1_ be the largest multiple of 10 such that _n_ - _n1_ is less than 10<sup>_precision_</sup>.
+      1. Let _v1_ be _v_ - (_n1_ / 10<sup>_precision_</sup>).
+      1. Let _N_ be ? EnsureDecimal128Value(_v1_, _precision_, *"halfEven"*).
+      1. Return Decimal128ToDecimalString(_N_).
     </emu-alg>
+    <emu-note>
+      <p>This operation follows the specification of the conversion of IEEE 754 Decimal128 values to strings (external character sequences) discussed in Section 5.12 of <emu-xref href="#sec-bibliography">IEEE 754-2019</emu-xref>.</p>
+    </emu-note>
   </emu-clause>
 
   <emu-clause id="sec-decimal.prototype.tobigint">
     <h1>Decimal128.prototype.toBigInt ( )</h1>
```

Comment:
Should this be a separate method, or an overloading of the BigInt constructor? (IMO this question shouldn't block Stage 2)

Reply:
I wasn't clear on how to do this -- if we're overloading the BigInt constructor, we can drop this, of course. I'm not sure which way to go. I'm happy with dropping one approach or the other.

Reply:
Oh I see, you already do overload the BigInt constructor. Yes, I think this method should be dropped (at the cost of polyfilling being more difficult)

Reply:
Sounds good. We can also drop the `toNumber` method for the same reason.

Reply:
done in #144